sympy/matrices/matrices.py

```diff
@@ -1581,7 +1585,12 @@ def _matrix_pow_by_recursion(self, num):
     def __pow__(self, num):
         if not self.is_square:
             raise NonSquareMatrixError()
-        if isinstance(num, (int, Integer)):
+        num = sympify(num)
+        if num.is_Number and num % 1 == 0:
```

Comment:
[PEP-8](https://www.python.org/dev/peps/pep-0008/#id30) says
> Block comments generally apply to some (or all) code that follows them, and are indented to the same level as that code.

I believe this means that this if-line should come *after* the block comment.